app/controllers/sessions_controller.rb

```diff
@@ -48,6 +48,16 @@ def account_manager_url
     Plek.find("account-manager")
   end
 
+  def fetch_http_referrer
+    http_referrer = request.headers["HTTP_REFERER"]
+
+    return nil unless http_referrer
+
+    return nil unless http_referrer&.start_with?(Plek.new.website_root)
```

Comment:
With the `&.` in this conditional, we don't need the `return nil unless http_referrer` above.  Also `return nil` is the same as `return`, so these three lines can be

```ruby
return unless http_referrer&.start_with?(Plek.new.website_root)
```

Reply:
ah great point, removed.